### DISCUSSION TITLE
Send Email to users when the token change.

### DISCUSSION THREAD

**Discussion by asoMansoury** [vpnhood/VpnHood#292] [General]
I think it is very good when the token of any clients will be changed, send the new tokens to them by email.
It is a good feature for countries which has strong filtering.
Another feature that I feel is necessary is about grouping client for any time the admin want to change for example their max traffic.

**Comment by bobvhood** [comment]
>send the new tokens to them by email.

We need to think much more about it, there are some other ideas too.

> Another feature that I feel is necessary is about grouping client for any time the admin want to change for example their max traffic.

Group update is in progress and we are working on it

**Reply by asoMan1369** [reply]
About sending email, believe me it is much more important than having groups.
it is critical when we are forced to shut down a server and change users' server. by having this feature it would send new token to clients automatically.

**Comment by bobvhood** [comment]
@asoMan1369 
thank you for your suggestion. We understand the importance of keeping our clients' tokens secure and appreciate your concern.
We are currently working on improving our system to make it more efficient and secure, and one of our plans is to implement a shared server farm key that would make it easier to update tokens and email for all users.

Unfortunately, we are not able to send emails to clients in our free plan on the access server. In the meantime, we recommend you to do it manually.

Thank you for your understanding and continued support.
